odoo/modules/registry.py

```diff
@@ -467,23 +467,31 @@ def check_indexes(self, cr, model_names):
                    [tuple(row[0] for row in expected)])
         existing = {row[0] for row in cr.fetchall()}
 
-        if not self.has_trigram and any(row[3] == 'trigram' for row in expected):
+        if not self.has_trigram and any(row[2].index == 'trigram' for row in expected):
             self.has_trigram = sql.install_pg_trgm(cr)
 
-        for indexname, tablename, columnname, index in expected:
+        for indexname, tablename, field in expected:
+            columnname = field.name
+            index = field.index
             assert index in ('btree', 'btree_not_null', 'trigram', True, False, None)
-            if index and indexname not in existing:
-                method = 'btree'
-                operator = ''
-                where = ''
-                if index == 'btree_not_null':
-                    where = f'"{columnname}" IS NOT NULL'
-                elif index == 'trigram' and self.has_trigram:
+            if index and indexname not in existing and \
+                    ((not field.translate and index != 'trigram') or (index == 'trigram' and self.has_trigram)):
+
+                if index == 'trigram':
+                    if not field.translate:
+                        expression = f'"{columnname}" gin_trgm_ops'
+                    else:  # field.translate == True
+                        # TODO VSC : see if we can simplify the index so it doesn't contain the [ and " and spaces, only values
+                        # TODO CWG: also check logic for inactive languages
+                        expression = f'(jsonb_path_query_array("{columnname}", \'$.*\'::jsonpath)::text) gin_trgm_ops'
```

Comment:
escaping quotes makes strings much harder to read. You could use triple-quotes to avoid that when you have to mix quote types (everywhere)